src/config.rs

```diff
@@ -81,6 +83,40 @@ impl From<PreAmplifier> for ffi::AudioProcessing_Config_PreAmplifier {
     }
 }
 
+/// General level adjustment in the capture pipeline.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "derive_serde", derive(Serialize, Deserialize))]
+pub struct CaptureLevelAdjustment {
+    /// Scales the signal before any processing is done.
+    pub pre_gain_factor: f32,
+
+    /// Scales the signal after all processing is done.
+    pub post_gain_factor: f32,
+}
+
+impl Default for CaptureLevelAdjustment {
+    fn default() -> Self {
+        Self { pre_gain_factor: 1.0, post_gain_factor: 1.0 }
+    }
+}
+
+impl From<CaptureLevelAdjustment> for ffi::AudioProcessing_Config_CaptureLevelAdjustment {
+    fn from(other: CaptureLevelAdjustment) -> Self {
+        let analog_mic_gain_emulation =
+            ffi::AudioProcessing_Config_CaptureLevelAdjustment_AnalogMicGainEmulation {
+                enabled: false,
+                initial_level: 255,
+            };
+
+        Self {
+            enabled: true,
+            pre_gain_factor: other.pre_gain_factor,
+            post_gain_factor: other.post_gain_factor,
+            analog_mic_gain_emulation,
+        }
+    }
+}
```

Comment:
Optional nit: we could instead implement `impl From<`**`Option`**`<CaptureLevelAdjustment>> for ffi::AudioProcessing_Config_CaptureLevelAdjustment {` and set `enabled: false` if it is None - i.e. move some logic from the caller to here.

OTOH I guess this pattern is used for many types here, so we shuold probably either convert either all or none - which is better suited for a separate PR.